examples/src/main/java/org/bitcoinj/examples/ForwardingService.java

```diff
@@ -107,45 +128,59 @@ public void start() {
         // Download the blockchain and wait until it's done.
         kit.startAsync();
         kit.awaitRunning();
+        wallet = kit.wallet();
+    }
+
+    /**
+     * @return The current receiving address of the forwarding wallet
+     */
+    public Address receivingAddress() {
+        return wallet.currentReceiveAddress();
     }
 
     /**
-     * Setup the listener to forward received coins and wait
+     * Setup a listener that will forward received coins and return a transaction
```

Comment:
We need more comments here. This method is the heart of the forwarding service and a detailed explanation of the asynchronous pipeline would be helpful -- especially in a sample app.